Userland/Applications/Browser/CookieJar.cpp

```diff
@@ -48,6 +49,26 @@ void CookieJar::set_cookie(const URL& url, Web::Cookie::ParsedCookie const& pars
     purge_expired_cookies();
 }
 
+void CookieJar::update_cookie(URL const& url, Web::Cookie::Cookie cookie)
+{
+    auto domain = canonicalize_domain(url);
+    if (!domain.has_value())
+        return;
+
+    if (!domain_matches(domain.value(), cookie.domain))
+        return;
+
+    CookieStorageKey key { cookie.name, cookie.domain, cookie.path };
+
+    if (auto old_cookie = m_cookies.find(key); old_cookie != m_cookies.end()) {
+        cookie.creation_time = old_cookie->value.creation_time;
```

Comment:
That the creation time has to stay the same is non-obvious, perhaps a link/spec text from 
https://www.rfc-editor.org/rfc/rfc6265#section-5.3 step 11.3 is nice?

Also that says that something about HttpOnly which I don't know if this follows